src/io/network/endpoint.cpp

```diff
@@ -40,26 +41,24 @@ Endpoint::IpFamily Endpoint::GetIpFamily(const std::string &ip_address) {
 std::optional<std::pair<std::string, uint16_t>> Endpoint::ParseSocketOrIpAddress(
     const std::string &address, const std::optional<uint16_t> default_port = {}) {
   /// expected address format:
-  ///   - "ip_address:port_number"
-  ///   - "ip_address"
+  ///   - "address:port_number"
+  ///   - "address"
+  /// An address can be either an IP address or a DNS-associated name with an IP address.
   /// We parse the address first. If it's an IP address, a default port must
   // be given, or we return nullopt. If it's a socket address, we try to parse
   // it into an ip address and a port number; even if a default port is given,
   // it won't be used, as we expect that it is given in the address string.
   const std::string delimiter = ":";
-  std::string ip_address;
+  std::string parsed_address;
 
   std::vector<std::string> parts = utils::Split(address, delimiter);
   if (parts.size() == 1) {
     if (default_port) {
-      if (GetIpFamily(address) == IpFamily::NONE) {
-        return std::nullopt;
-      }
-      return std::pair{address, *default_port};
+      return std::pair{parsed_address, *default_port};
```

Comment:
I removed this check because the same check is performed later after creating the Endpoint structure in replication.cpp:160.